graph/src/components/subgraph/instance.rs

```diff
@@ -7,6 +7,7 @@ pub struct DataSourceTemplateInfo {
     pub data_source: String,
     pub template: DataSourceTemplate,
     pub params: Vec<String>,
+    pub context: Option<Entity>,
```

Comment:
Similar to the `Entity` vs. `DataSourceContext` comment in the graph-ts review, I think even for us it'll be confusing to see `Entity` here, say, six months down the road. My suggestion is to add a `DataSourceContext` type here as well.